MeesGame/Gameplay/GameObjects/TilesTypes/DoorTile.cs

```diff
@@ -31,51 +32,41 @@ public Color SecondarySpriteColor
             }
         }
 
+        private bool IsComingFromAllowedDirection(ITileFieldPlayer player)
+        {
+            if (isHorizontal)
+                // A player is allowed to move onto a horizontalDoor from the tile above or below it.
+                return player.Location.X == Location.X;
+            else
+                // A player is allowed to move onto a verticalDoor from the Tile to the left or to the right of it.
+                return player.Location.Y == Location.Y;
+        }
+
         public override bool CanPlayerMoveHere(ITileFieldPlayer player)
         {
-            // A player is allowed to move onto a horizontalDoor from the tile above or below it.
-            // A player is allowed to move onto a verticalDoor from the Tile to the left or to the right of it.
-            if ((isHorizontal && player.Location.X == Location.X) || (!isHorizontal && player.Location.Y == Location.Y))
-            {
-                //A player is allowed to move onto a door tile if the door is open.
-                if (CanMoveOntoDoor(player))
-                    return true;
-                else
-                    return false;
-            }
-            else return false;
+            //A player is allowed to move onto a door tile if he comes from the correct direction and 
+            //the door is open.
+            return IsComingFromAllowedDirection(player) && IsDoorOpenFor(player);
```

Comment:
It should check if it can move onto the next tile, just like the floortile does (why was that check removed from the player?). It would be even better if the standard logic was moved the the abstract tile class and this called base.CPMH after doing it's own checks.